src/workloads/scale/ReadMemoryStressUntilFailure.yml

```diff
@@ -0,0 +1,131 @@
+SchemaVersion: 2018-07-01
+Owner: Service Architecture
+Description: |
+  Used to test performance of the server under heavy memory pressure caused by read
+  operations. This workload is expected to fail due to host(s) being unreachable as a
+  result of mongod(s) running out of memory.
+
+  To achieve this, documents are inserted in the form of {a: <id>, b: <random number>,
+  c: <fill 520 KB>}. Many threads are then spawned to run aggregate sort on a number of
+  documents. The document size and the number of documents to sort by each thread are
+  tuned to be just under 100 MB, which is the memory limit for operations before
+  spilling to disk. The threads would therefore each cause a close-to-max amount of
+  memory to be used. Increasing the number of threads should cause the host(s) that
+  process the operations to fail due to out-of-memory errors.
+
+Keywords:
+- scale
+- memory stress
+- aggregate
+- sort
+- insert
+- fail
+- oom
+- out of memory
+
+GlobalDefaults:
+  dbname: &DBName memorystress
+  MaxPhases: &MaxPhases 20
+
+  LoadThreads: &LoadThreads 128
+  LoadBatchSize: &LoadBatchSize 500
+
+  NumberOfDocuments: &NumDocs 20096
+  # NumberOfDocumentsPerWorker = NumberOfDocuments / LoadThreads.
+  NumberOfDocumentsPerWorker: &DocsPerThread 157
+  # Used in Document: a to let the starting value of a = 0.
+  DocumentStartAt: &DocStartAt -157
+  DocSize: &DocSize 520000
+  Document: &Doc
+    # Document number/ID. Starts at *DocStartAt = -*DocsPerThread = -157 because ActorId
+    # starts at 1, and therefore in order for the first output of ^Inc to be 0, "start"
+    # would need to be offset by ActorId * *DocsPerThread = 157.
+    a: {^Inc: {start: *DocStartAt, step: 1, multiplier: *DocsPerThread}}
+    # Random number from [0, NumberOfDocuments].
+    b: {^RandomInt: {min: 0, max: *NumDocs}}
+    # Fill 520 KB.
+    c: {^FastRandomString: {length: *DocSize}}
+
+  # Number of documents to sort by each thread. Tuned so that *SortStep * *DocSize < 100 MB,
+  # which is the maximum memory a sort query is allowed to use before spilling to disk.
+  SortStep: &SortStep 180
+  SortBatchSize: &SortBatchSize 180
+  SortThreads: &SortThreads 250
+  SortRepeat: &SortRepeat 5
+
+Clients:
+  Default:
+    QueryOptions:
+      socketTimeoutMS: -1
```

Comment:
what does a socketTimeoutMs of -1 mean?

Reply:
I assumed many workloads use -1 as max timeout but I can't find proof of that this is max timeout, will look for it